syntax = "proto3";
option cc_enable_arenas = true;

package Ydb.Debug;
option java_package = "com.yandex.ydb.debug";
option java_outer_classname = "DebugProtos";

import "ydb/public/api/protos/ydb_issue_message.proto";
import "ydb/public/api/protos/ydb_status_codes.proto";

// just go to GRPC without the rest of YDB

message PlainGrpcRequest {
}

message PlainGrpcResponse {
    StatusIds.StatusCode status = 1;
    repeated Ydb.Issue.IssueMessage issues = 2;
}

// Go until GrpcProxy

message GrpcProxyRequest {
}

message GrpcProxyResponse {
    StatusIds.StatusCode status = 1;
    repeated Ydb.Issue.IssueMessage issues = 2;
}

// Ping KQP proxy without executing anything

message KqpProxyRequest {
}

message KqpProxyResponse {
    StatusIds.StatusCode status = 1;
    repeated Ydb.Issue.IssueMessage issues = 2;
}

// Ping SchemeCache

message SchemeCacheRequest {
}

message SchemeCacheResponse {
    StatusIds.StatusCode status = 1;
    repeated Ydb.Issue.IssueMessage issues = 2;
}

// Ping TxProxy

message TxProxyRequest {
}

message TxProxyResponse {
    StatusIds.StatusCode status = 1;
    repeated Ydb.Issue.IssueMessage issues = 2;
}

// Ping Actor Chain

message ActorChainRequest {

    // number of actors to be created, default 10
    uint32 ChainLength = 1;

    // immitate work duration for each actor (approximate), default ~ 5 usec
    uint32 WorkUsec = 2;

    // don't use tail sends and registrations
    bool NoTailChain = 3;
}

message ActorChainResponse {
    StatusIds.StatusCode status = 1;
    repeated Ydb.Issue.IssueMessage issues = 2;
}
